src/main.rs

```diff
@@ -1,4 +1,147 @@
+        .route("/:version/contacts", get(contacts_handler))
+        .route("/:version/dictionary", get(dictionary_handler))
+        .route("/:version/faqs", get(faqs_handler))
+        .route("/:version/food/named/menu/pause", get(pause_menu_handler))
+        .route(
+            "/:version/printing/color-printers",
+            get(color_printers_handler),
+        )
+        .route("/:version/spaces/hours", get(hours_handler))
+        .route("/:version/tools/help", get(help_handler))
+        .route("/:version/transit/bus", get(transit_bus_handler))
+        .route("/:version/transit/modes", get(transit_modes_handler))
+        .route("/:version/webcams", get(webcams_handler))
+        .fallback(fallback.into_service());
+
+    Server::bind(&"0.0.0.0:3000".parse().unwrap())
+        .serve(app.into_make_service())
+        .await
+        .unwrap();
+}
+
+#[allow(clippy::unused_async)]
+async fn root_handler() -> Result<&'static str, StatusCode> {
+    Ok("Hello world!")
+}
+
+#[allow(clippy::unused_async)]
+async fn heartbeat_handler() -> Result<&'static str, StatusCode> {
+    Ok("pong")
+}
+
+#[allow(clippy::unused_async)]
+async fn fallback() -> impl IntoResponse {
+    (StatusCode::NOT_FOUND, "Not Found".to_string())
+}
+
+async fn contacts_handler(_version: Version) -> Result<Json<Value>, StatusCode> {
+    let data = gh_pages_handler("contact-info.json").await.unwrap();
+    Ok(data)
+}
+
+async fn dictionary_handler(_version: Version) -> Result<Json<Value>, StatusCode> {
+    let data = gh_pages_handler("dictionary.json").await.unwrap();
+    Ok(data)
+}
+
+async fn faqs_handler(_version: Version) -> Result<Json<Value>, StatusCode> {
+    let data = gh_pages_handler("faqs.json").await.unwrap();
+    Ok(data)
+}
+
+async fn pause_menu_handler(_version: Version) -> Result<Json<Value>, StatusCode> {
+    let data = gh_pages_handler("pause-menu.json").await.unwrap();
+    Ok(data)
+}
+
+async fn color_printers_handler(_version: Version) -> Result<Json<Value>, StatusCode> {
+    let data = gh_pages_handler("color-printers.json").await.unwrap();
+    Ok(data)
 }
 
+async fn hours_handler(_version: Version) -> Result<Json<Value>, StatusCode> {
+    let data = gh_pages_handler("building-hours.json").await.unwrap();
+    Ok(data)
+}
+
+async fn help_handler(_version: Version) -> Result<Json<Value>, StatusCode> {
+    let data = gh_pages_handler("help.json").await.unwrap();
+    Ok(data)
+}
+
+async fn transit_bus_handler(_version: Version) -> Result<Json<Value>, StatusCode> {
+    let data = gh_pages_handler("bus-times.json").await.unwrap();
+    Ok(data)
+}
+
+async fn transit_modes_handler(_version: Version) -> Result<Json<Value>, StatusCode> {
+    let data = gh_pages_handler("transportation.json").await.unwrap();
+    Ok(data)
+}
+
+async fn webcams_handler(_version: Version) -> Result<Json<Value>, StatusCode> {
+    let data = gh_pages_handler("webcams.json").await.unwrap();
+    Ok(data)
+}
```

Comment:
This pattern could be macro-ified if you want :wink: 

```suggestion
macro_rules! gh_pages_handler {
    ($name:ident,$filename:literal) => {
        async fn $name(_version: Version) -> Result<Json<Value>, StatusCode> {
            let data = gh_pages_handler($filename).await.unwrap();
            Ok(data)
        }
    };
}

gh_pages_handler!(contacts_handler, "contact-info.json");
gh_pages_handler!(dictionary_handler, "dictionary.json");
gh_pages_handler!(faqs_handler, "faqs.json");
gh_pages_handler!(pause_menu_handler, "pause-menu.json");
gh_pages_handler!(color_printers_handler, "color-printers.json");
gh_pages_handler!(hours_handler, "building-hours.json");
gh_pages_handler!(help_handler, "help.json");
gh_pages_handler!(transit_bus_handler, "bus-times.json");
gh_pages_handler!(transit_modes_handler, "transportation.json");
gh_pages_handler!(webcams_handler, "webcams.json");
```

Reply:
And, if you want to get _really_ fancy, this pattern can be DRYed even further:

```suggestion
macro_rules! gh_pages_handler {
    ($name:ident,$filename:literal) => {
        async fn $name(_version: Version) -> Result<Json<Value>, StatusCode> {
            let data = gh_pages_handler($filename).await.unwrap();
            Ok(data)
        }
    };
}

macro_rules! gh_pages_handlers {
    ($([$name:ident, $filename:literal]),+) => {
        $(
            gh_pages_handler!($name, $filename);
        )+
    };
}

gh_pages_handlers!(
    [contacts_handler, "contact-info.json"],
    [dictionary_handler, "dictionary.json"],
    [faqs_handler, "faqs.json"],
    [pause_menu_handler, "pause-menu.json"],
    [color_printers_handler, "color-printers.json"],
    [hours_handler, "building-hours.json"],
    [help_handler, "help.json"],
    [transit_bus_handler, "bus-times.json"],
    [transit_modes_handler, "transportation.json"],
    [webcams_handler, "webcams.json"] // I'm not aware of a way to make `macro_rules` ignore a final comma.
);
```

Reply:
Oh, actually, you can allow optional trailing commas by adding a `$(,)?` thing to the `($([$name:ident, $filename:literal]),+)`:

```rust
macro_rules! gh_pages_handlers {
    ($([$name:ident, $filename:literal]),+ $(,)?) => {
        // ...
    };
}
```

Reply:
That _is_ fancy!

I am now trying to picture the next step after this macro. 

What do you think about there being structs for each return type so that we have some guarantee of what the return types are, that they can be serialized and deserialized, and for us to provide exported typedefs for clients?

I am not entirely sure as of yet how to implement beyond defining the types, but we're taking it in steps.

Reply:
Structs for each return type is a great idea and ultimately how we'd probably want to encode each type.

So, for example, `FaqsResponse`, `BusTimesResponse`, `OtherModesResponse`, etc.

(For what it's worth, you can modify the macro to include a `$response_type:ty` if you need a way of specifying what the expected return type is for a given instance of the `gh_pages_handler` macro — if you define an example return type for one of our handlers, I can play around with it to see if we can get the handling correct.)